src/Storefront/Page/Product/Review/ReviewLoaderResult.php

```diff
@@ -5,97 +5,197 @@
 use Shopware\Core\Content\Product\Aggregate\ProductReview\ProductReviewCollection;
 use Shopware\Core\Content\Product\Aggregate\ProductReview\ProductReviewEntity;
 use Shopware\Core\Content\Product\SalesChannel\Review\RatingMatrix;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult;
+use Shopware\Core\Framework\Feature;
 use Shopware\Core\Framework\Log\Package;
 use Shopware\Storefront\Framework\Page\StorefrontSearchResult;
 
-/**
- * @template-extends StorefrontSearchResult<ProductReviewCollection>
- */
-#[Package('storefront')]
-class ReviewLoaderResult extends StorefrontSearchResult
-{
+if (Feature::isActive('v6.7.0.0')) {
```

Comment:
Hi @aragon999, am pasting here comments from the internal core team review:

You can do this, but you shouldn't. The use of the FeatureFlags in this MR are not ideal, as they try to change everything completely on the point of v6.7, this will most likely fail on the 6.7 mayor release because this code only runs in tests for months. The aproach should be made more integrational.

Reply:
Mh, but how should the change of the type `StorefrontSearchResult` -> `EntitySearchResult` be implemented otherwise?

Reply:
The `StorefrontSearchResult` already extends from `EntitySearchResult`, so there is no need for a hard rewrite on this step. The use of `StorefrontSearchResult` methods in 6.7 is already captured by the triggerDeprecation methods. So I would just leave the `ReviewLoaderResult` techically as it is and just add a deprecation notice, that it will be extended directly from `EntitySearchResult` from 6.7 on,

```suggestion
* deprecated tag:v6.7.0 - will be extended directly from `EntitySearchResult`.
class ReviewLoaderResult extends StorefrontSearchResult /* extends EntitySearchResult */
{
```

Reply:
@ssltg thank you for the feedback. But in general I meant more the part of the return types:
```php
    public function getReviews(): EntitySearchResult
    {
        return $this->reviews;
    }
```
vs
```php
    public function getReviews(): StorefrontSearchResult
    {
        return $this->reviews;
    }
```

Should I also do it with comments there? But than I do not know how to implement it in the rest of the code?

Reply:
@aragon999 ok, I see your problem here. The reviews are a bit odd implemented here in the first place. Leave the code like it is. Sorry for the inconveniences.

Reply:
@ssltg okay very good :-) Yes it is also implemented the same way in other places, therefore I did not find another solution which was non-breaking.

Regarding the reviews, yes, I once tried to clean that a bit up, but it was also quite challenging: https://github.com/shopware/shopware/pull/2389/files

Maybe once this is merged, I can give this another try in the Shopware 6.6 cycle.